### PR TITLE
Accept int32 as index for index_add op

### PR DESCRIPTION
**Pull request by JackCaoG**
This is to fix https://github.com/pytorch/xla/issues/2575, in pytorch pr https://github.com/pytorch/pytorch/pull/46758, `index_add` with `int32` support is added, xla needs to follow the same logic.

### CODE REVIEW

**Comment by ailzhang** [comment]
could you add torch_pin to check whether this PR fixes all issues from the upstream PR? Thanks!

**Comment by JackCaoG** [comment]
> could you add torch_pin to check whether this PR fixes all issues from the upstream PR? Thanks!

sure, I run it locally(with pytorch patch) and confirmed that failed test is passing with this change.

**Review by qizzzh — APPROVED** [review]

**Comment by qizzzh** [comment]
Thanks for the fix! I'm updating index_select to support int32 indices as well. Do you think we need to fix that for XLA too?

**Comment by JackCaoG** [comment]
@qizzzh qi will that be in a separate pr? If that's the case we can do a follow up pr on pt/xla when that happens.

**Comment by JackCaoG** [comment]
test passed locally on my cpu machine, rerun it on circleCI

**Comment by qizzzh** [comment]
> @qizzzh qi will that be in a separate pr? If that's the case we can do a follow up pr on pt/xla when that happens.

It's in the same PR. Just updated. Please feel free to take a look. Thanks!

File: torch_xla/csrc/ops/index_ops.cpp [hunk 1]
```diff
@@ -310,8 +310,9 @@ ir::NodePtr IndexFill(const XLATensor& base, xla::int64 dim,
 
 ir::Value IndexAdd(const XLATensor& base, xla::int64 dim,
                    const XLATensor& index, const XLATensor& source) {
-  XLA_CHECK_EQ(index.dtype(), at::ScalarType::Long)
-      << "Add index is expected to be of scalar type Long";
+  XLA_CHECK(index.dtype() == at::ScalarType::Long ||
+            index.dtype() == at::ScalarType::Int)
+      << "Add index is expected to be of scalar type Long or scalar type Int";
```
Comment (davidel) [inline torch_xla/csrc/ops/index_ops.cpp:315]: Maybe we can pipe `index.dtype()` into the error stream to give some hint?

**Review by davidel — COMMENTED** [review]
This needs a test ...

**Comment by JackCaoG** [comment]
@qizzzh it seems like pt/xla does not check `index_select` index type, so no change needs to be done.

**Review by davidel — APPROVED** [review]

**Pull request merged**